src/commonmark/commonmarkdataprocessor.js

```diff
@@ -158,6 +158,20 @@ export default class CommonMarkDataProcessor {
 			}
 		});
 
+		turndownService.addRule( 'mentions', {
+			filter: (node) => {
+				return (
+					node.nodeName === 'MENTION' &&
```

Comment:
Perhaps the filter should go for the nodeName only, and then the part of the application that depends on the `data-` attributes should error if they are missing. I don't think the processor is responsible for checking the validity.

However, no matter which part is responsible for checking, I think it should throw an error and not just silently ignore the element.

Reply:
Hi @b12f ,
Thank you for your feedback.

You are right, it doesn't make sense to check for these attributes since it is guaranteed that the user will always have id and name and we are building the <mention> tag right in the front with this data. I've removed them, I just kept the class (node.nodeName === 'MENTION' && node.classList.contains('mention')) to avoid any string matching <mention></mention> by chance.

It seems to me that the responsibility for the integrity of the user object would be on the backend and the user edit forms.